google-login-plugin/src/com/google/gct/login/stats/UsageTrackerManager.java

```diff
@@ -0,0 +1,50 @@
+package com.google.gct.login.stats;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.diagnostic.Logger;
+
+/**
+ * Stores the users choice to opt in/out of sending usage metrics via the Google Usage Tracker.
+ */
+public class UsageTrackerManager {
+    private static final Logger LOG = Logger.getInstance(UsageTrackerManager.class);
+    private static final String USAGE_TRACKER_KEY = "GOOGLE_CLOUD_TOOLS_USAGE_TRACKER_OPT_IN";
+    private static final String USAGE_TRACKER_PROPERTY = "usage.tracker.property";
+
+    private UsageTrackerManager() {}
+
+    public static void setOptIn(boolean optIn) {
+        PropertiesComponent.getInstance().setValue(USAGE_TRACKER_KEY, String.valueOf(optIn));
+    }
+
+    public static boolean getOptIn() {
+        return PropertiesComponent.getInstance().getBoolean(USAGE_TRACKER_KEY, false);
+    }
+
+    /**
+     * @return true if the user has opted in/out of usage tracking; false otherwise
+     */
+    public static boolean isTrackingConfiguredByUser() {
```

Comment:
getOptIn() tells you whether the user has opted in/out of using our usage tracking system (so true means the agreed to send usage analytics to Google and false if they declined) . 
isTrackingConfiguredByUser() tells you if the user has ever agreed or disagreed to send analytics to Google. For example once the user installs our plugin and restarts IntelliJ, they would never have agreed/disagreed to our usage tracking system, so is  isTrackingConfiguredByUser() will return true.